blueflood-core/src/main/java/com/rackspacecloud/blueflood/eventemitter/RollupEventEmitter.java

```diff
@@ -41,10 +52,30 @@ private RollupEventEmitter() {
     @Override
     public Future emit(final String event, final RollupEvent... eventPayload) {
         //TODO: This hack will go away after Kafka Serializer is made generic
-        if(eventPayload[0].getRollup() instanceof BasicRollup) {
+        if(eventPayload[0].getRollup() instanceof BasicRollup && super.hasListeners(ROLLUP_EVENT_NAME)) {
             return eventExecutors.submit(new Runnable() {
                 @Override
                 public void run() {
+                    if (Configuration.getInstance().getBooleanProperty(CoreConfig.USE_ES_FOR_UNITS) && Configuration.getInstance().getListProperty(CoreConfig.DISCOVERY_MODULES).contains(Util.ElasticIOPath)) {
```

Comment:
In some cases you check for 
```
Configuration.getInstance().getListProperty(CoreConfig.DISCOVERY_MODULES).contains(Util.ElasticIOPath)) 
```

In some other you do not. Do we need it everywhere?